### DISCUSSION TITLE
CanelGesture()실행후 다시 SetGesture()했을때 애니메이션이 실행이 안됩니다.

### DISCUSSION THREAD

**Discussion by jiwoo50** [naverz/zepeto-studio-kor#1163] [5. Q&A]
애니메이션 클립의 loop설정을 켜고 setGesture를 한번하고 cancelGesture로 취소하고 다시 setGesture를 실행하면 애니메이션 실행이 안되고 원래의 idle상태로 돌아가는데 혹시 해결할 방법이 없나요?
loop를 끄면 해결되긴 하는데 너무 재생시간이 짧아서 별도의 행동을 하기전까지 계속 애니메이션을 실행하고 싶습니다.

**Comment by smlee2952** [comment]
로직이 제대로 흘러갔는지 확인이 필요할 것 같습니다.

저도 애니메이션의 loop를 키고 사용하고 있는데 문제 없습니다.
cancelGesture를 호출하면 이미 캐릭터는 idle 상태로 돌아가있어야 합니다.

로그를 추가하셔서 확인한번 해보세요!

**Comment by zlmrzlo** [comment]
cancelGesture와  setGesture 사이에 텀을 좀 주는 것은 어떨까요?
바로 붙어있어서 그 과정에서 오류가 생길 수도 있을 것 같아요.

**Reply by smlee2952** [reply]
이게 맞는 것 같습니다.

제 경험 상 제페토 캐릭터가 제공하는 함수를 연속으로 부르면 정상동작 하지 않았습니다.
그래서 텔레포트 후 제스쳐 기능을 구현할 때 텔레포트를 먼저 하게 하고 제스쳐는 코루틴으로 돌렸습니다.
